include/bm/bm_sim/data.h

```diff
@@ -180,28 +180,28 @@ class Data {
            typename std::enable_if<std::is_integral<T>::value, int>::type = 0>
   T get() const {
     assert(arith);
-    return static_cast<T>(value);
+    return value.convert_to<T>();
```

Comment:
based on the compiler errors in the regression, you may need to apply `std::remove_const` to `T`